### PR TITLE
SWATCH-294 Lookup RHM service via orgId rather than account number wh…

### PR DESCRIPTION
**Pull request by kartikshahc**
…en present

### CODE REVIEW

**Comment by kartikshahc** [comment]
https://issues.redhat.com/browse/SWATCH-294

1. **Description**:
- Currently we pass accountnumber as a JSON key in RhMarketplaceJmxBean>rhMarketplaceJmxBean>submitTallySummary. Since we are replacing accountnumber with orgId. The service should still find the data and send it to RHM service. also, if we have only accountnumber then it will fetch orgId and should return same data. Only when accountnumber doesn't have an orgId in that case it will check by accountnumber but, this case won't be valid going further.

2. **Test Steps**:
- Start app with `DEV_MODE=true SUBSCRIPTION_USE_STUB=true USER_USE_STUB=true ./gradlew :bootRun`

- Insert Mock data:
```
INSERT INTO public.offering(
    sku, product_name, product_family, sla, usage, description, has_unlimited_usage)
    VALUES ('MW01882', 'OpenShift Streams for Apache Kafka', 'JBoss', 'Premium', 'Production', 'Red Hat OpenShift Streams for Apache Kafka (Hourly)', false);

INSERT INTO public.subscription(
	sku, owner_id, subscription_id, quantity, start_date, end_date, billing_provider_id, account_number, subscription_number, billing_provider, billing_account_id)
	VALUES ('MW01882', 'org123', '12351253', '1', '2012-05-13 16:32:00+00', '2032-05-13 16:32:00+00', '1', 'account123', '1235153', 'red hat', '456');
```

- Execute JMX endpoint RhMarketplaceJmxBean>rhMarketplaceJmxBean>submitTallySummary

1. Json object with orgId:

```
{     "org_id":"org123",   "tally_snapshots": [         {             "id": "c204074d-626f-4272-aa05-b6d69d6de16a",     "billing_provider": "red hat",  "billing_account_id": "1",       "snapshot_date": 1615406400,             "product_id": "rhosak",             "usage": "Production",             "sla": "Premium",             "granularity": "Hourly",             "tally_measurements": [                 {                     "hardware_measurement_type": "TOTAL",                     "uom": "Cores",                     "value": 36                 },                 {                     "hardware_measurement_type": "PHYSICAL",                     "uom": "Cores",                     "value": 36                 }             ]         }     ] }
```

2. Json object with accountnumber:
```
{     "account_number": "account123",     "tally_snapshots": [         {             "id": "c204074d-626f-4272-aa05-b6d69d6de16a",     "billing_provider": "red hat",  "billing_account_id": "1",       "snapshot_date": 1615406400,             "product_id": "rhosak",             "usage": "Production",             "sla": "Premium",             "granularity": "Hourly",             "tally_measurements": [                 {                     "hardware_measurement_type": "TOTAL",                     "uom": "Cores",                     "value": 36                 },                 {                     "hardware_measurement_type": "PHYSICAL",                     "uom": "Cores",                     "value": 36                 }             ]         }     ] }
```

- For both the above json object we should get billingAccountId in response data. It will look like below in logs:
```
usageRequest to be sent: class UsageRequest {
    data: [class UsageEvent {
        start: 1615406400000
        end: 1615410000000
        subscriptionId: 1
        eventId: c204074d-626f-4272-aa05-b6d69d6de16a
        additionalAttributes: {}
        measuredUsage: [class UsageMeasurement {
            chargeId: null
            metricId: null
            value: 36.0
        }]
    }]
}
```

File: src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java [hunk 1]
```diff
@@ -547,8 +546,13 @@ public List<org.candlepin.subscriptions.db.model.Subscription> findSubscriptions
             .billingAccountId(usageKey.getBillingAccountId())
             .payg(true)
             .beginning(rangeStart)
-            .ending(rangeEnd)
-            .build();
+            .ending(rangeEnd);
+
+    ReportCriteria subscriptionCriteria =
+        orgId
+            .map(s -> reportCriteriaBuilder.orgId(s).build())
```
Comment (kflahert) [inline src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java:553]: Super minor gripe, I think we should stick to using more informational variable names in lambdas. Could you rename 's' to 'id' or something similar?

Comment (kartikshahc) [reply src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java]: Good suggestion. I have addressed it.

**Review by kflahert — APPROVED** [review]

**Pull request merged**